pkg/network/master/subnets.go

```diff
@@ -55,15 +55,68 @@ func (master *OsdnMaster) SubnetStartMaster(clusterNetworks []common.ClusterNetw
 	}
 	master.subnetAllocatorList = subnetAllocatorList
 
+	err = master.verifyHostSubnetsState(subnets.Items)
+	if err != nil {
+		return err
+	}
+
 	master.watchNodes()
 	master.watchSubnets()
 	return nil
 }
 
+func (master *OsdnMaster) verifyHostSubnetsState(hostSubnets []networkapi.HostSubnet) error {
```

Comment:
The name perhaps should indicate that it can delete.  To me verify would not delete...

Perhaps "reconcileHostSubnets" ?